embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImpl.kt

```diff
@@ -56,24 +57,33 @@ class IntakeServiceImpl(
             service.store(metadata) { stream ->
                 serializer.toJson(intake, metadata.envelopeType.serializedType, stream)
             }
-            val lastCachedSession = lastCacheSessionRef
+            val lastReference = cacheReferences[metadata.envelopeType]
 
             if (metadata.complete) {
                 schedulingService.onPayloadIntake()
                 deliveryTracer?.onPayloadIntake(metadata)
-            } else if (metadata.envelopeType == SupportedEnvelopeType.SESSION) {
-                lastCacheSessionRef = metadata
+            } else {
+                cacheReferences[metadata.envelopeType] = metadata
+                if (!cacheableEnvelopeTypes.contains(metadata.envelopeType)) {
```

Comment:
shouldn't we check this before storing it to cacheReferences[]?

Reply:
We still want to cache it - we also want to know that we are because it's unexpected. I want to err on the side of this working and notifying us about it, rather than rejecting it.